docs/developers/multiprocessing.rst

```diff
@@ -0,0 +1,134 @@
+Parallelization via multiprocessing
+===================================
+
+Background
+----------
+
+In neutron imaging, the input data is often a 3D array.
+Depending on the axis of observation, the data can be interpreted as either a
+stack of radiographs (rotation angle) or a stack of sinograms (row number).
+Most of the filters in iMars3D is intended to adjust one single 2D image, therefore
+the filtering of a 3D array can be and should be parallelized over the axis that is
+not of interest.
+For instance, a typical radiograph from neutron imaging can have some contamination
+from stray gamma radiation, resulting in nearly/over saturated pixels randomly spreading
+over the image.
+Since the identification and correction of these gamma contamination does not depend on
+the rotation angle, the corresponding filter can be parallelized over the rotation angle
+axis so as to improve the overall performance.
+
+
+Usage of multiprocessing and real-time feedback via progress-bar
+----------------------------------------------------------------
+
+The modern Python have native support for parallelization, via either the
+`multithreading <https://docs.python.org/3/library/threading.html>`_ or
+the `multiprocessing <https://docs.python.org/3/library/multiprocessing.html#module-multiprocessing>`_ module.
+A high-level encapsulation package,
+`concurrent.futures <https://docs.python.org/3/library/concurrent.futures.html#module-concurrent.futures>`_,
+also becomes available in Python 3.2, which significantly simplifies the implementation of parallelization.
+Generally speaking, multithreading is useful for non-CPU intensive tasks as it does not entirely
+bypass the GIL (Global Interpreter Lock).
+In the case of iMars3D, multiprocessing is selected as most of the filters are computationally
+intensive.
+When using multiprocessing, the standard Numpy array is pickled and sent to the worker process via
```

Comment:
Let's drop the "how it used to be" portions of this. In a year nobody is going to care and it will bring up questions.